### PR TITLE
Fix imagine7 constants

### PR DESCRIPTION
**Pull request by joschi127**
Not so nice but working fix for https://github.com/avalanche123/Imagine/issues/496

Does anyone else know a better solution?

The constants seem to be gone. I could not find new constants to use.

### CODE REVIEW

**Comment by andris-sevcenko** [comment]
You can find the correct constants in the Imagick PECL [changelog](https://pecl.php.net/package-info.php?package=imagick&version=3.4.3RC1)

**Comment by takobell** [comment]
Looks like it's not just these enums that have changed, there are several others listed in the changelog that Imagine is using as well https://github.com/mkoppanen/imagick/issues/151#issuecomment-225014365 and https://pecl.php.net/package-info.php?package=imagick&version=3.4.3RC1

**Comment by joschi127** [comment]
thanks :+1:

**Comment by robfrawley** [comment]
@burzum We should add a `@todo` or other note in the comments, as hard coding these values *will* backfire at some point once they stop being supported. They were removed for a reason, so we should either figure out what constants replace them, or otherwise what functionality replaced them. Otherwise, this will come back to bite us at some version in the future.

Not against this merge or fix, but we need to document in a way that can be easily found via a regular expression or otherwise so we can locate places in the code that are possibly going to cause issues down the road.

File: lib/Imagine/Imagick/Image.php [hunk 1]
```diff
@@ -807,11 +807,17 @@ private function getMimeType($format)
      */
     private function setColorspace(PaletteInterface $palette)
     {
-        static $typeMapping = array(
```
Comment (romainneutron) [inline lib/Imagine/Imagick/Image.php]: why removing the static definition?

Comment (romainneutron) [inline lib/Imagine/Imagick/Image.php]: It removes memory optimization on PHP5

Comment (romainneutron) [inline lib/Imagine/Imagick/Image.php]: Fixed in 816a34d

**Comment by robfrawley** [comment]
This should probably be handled using the proper constants:

- `IMGTYPE_TRUECOLORMATTE` -> `IMGTYPE_TRUECOLORALPHA`
- `IMGTYPE_GRAYSCALEMATTE` -> `IMGTYPE_GRAYSCALEALPHA`

See https://github.com/Intervention/image/pull/679

**Comment by joschi127** [comment]
Also see my source code comment starting in line 813

https://github.com/joschi127/Imagine/blob/bbb008c076e167e414153a740c853433fd636754/lib/Imagine/Imagick/Image.php#L813

I think I have already tried to use the correct constants but these were not available as well. I do not know why. Maybe only with certain, maybe broken, versions of the Imagick extension or maybe I did something wrong. If you want we can try again and I can also test again on the server environment which was affected when implementing this fix.

**Comment by joschi127** [comment]
Also found these comments here:

So it sounds like  is broken with ImageMagick 7.0.4-6 and imagick 3.4.2_7 and fixed with imagick >= 3.4.3.

https://github.com/Intervention/image/pull/679#issuecomment-277670559

So if imagick >= 3.4.3 is now available on all stable distributions we can remove the work around with hard coded constants again?

**Pull request merged**